src/main/java/org/opensearch/flowframework/workflow/RegisterLocalModelStep.java

```diff
@@ -176,10 +155,80 @@ public void onFailure(Exception e) {
             if (functionName != null) {
                 mlInputBuilder.functionName(FunctionName.from(functionName));
             }
+            if (deploy != null) {
+                mlInputBuilder.deployModel(deploy);
+            }
 
             MLRegisterModelInput mlInput = mlInputBuilder.build();
 
-            mlClient.register(mlInput, actionListener);
+            mlClient.register(mlInput, ActionListener.wrap(response -> {
+                logger.info("Local Model registration task creation successful");
+
+                String taskId = response.getTaskId();
+
+                // Attempt to retrieve the model ID
+                retryableGetMlTask(
+                    currentNodeInputs.getWorkflowId(),
+                    currentNodeId,
+                    registerLocalModelFuture,
+                    taskId,
+                    "Local model registration",
+                    ActionListener.wrap(mlTask -> {
+
+                        // Registered Model Resource has been updated
+                        String resourceName = getResourceByWorkflowStep(getName());
+                        String id = getResourceId(mlTask);
+
+                        if (Boolean.TRUE.equals(deploy)) {
+
+                            // Simulate Model deployment step and update resources created
+                            flowFrameworkIndicesHandler.updateResourceInStateIndex(
+                                currentNodeInputs.getWorkflowId(),
+                                currentNodeId + "-deploy",
```

Comment:
Let's create a constant for the suffix.

Reply:
Actually, I'm not sure we need a suffix in the state if we're not using the map version anymore which previously had a key collision.

If we do keep a suffix I'd prefer a random one that is unlikely to match anything a user might have on a different step.

Reply:
That's a good point Dan, let me test out if the suffix is needed, if not then I'll remove this